app/controllers/v0/sign_in_controller.rb

```diff
@@ -109,12 +109,13 @@ def refresh
       raise SignIn::Errors::MalformedParamsError.new message: 'Refresh token is not defined' unless refresh_token
 
       decrypted_refresh_token = SignIn::RefreshTokenDecryptor.new(encrypted_refresh_token: refresh_token).perform
-      session_container = SignIn::SessionRefresher.new(refresh_token: decrypted_refresh_token,
-                                                       anti_csrf_token:).perform
-      serializer_response = SignIn::TokenSerializer.new(session_container:,
-                                                        cookies: token_cookies).perform
+      session_container = SignIn::SessionRefresher.new(refresh_token: decrypted_refresh_token, anti_csrf_token:).perform
+      serializer_response = SignIn::TokenSerializer.new(session_container:, cookies: token_cookies).perform
+      context = session_container.access_token.to_s
+                                 .merge({ type: session_container.session.user_verification.credential_type,
+                                          icn: session_container.session.user_account.icn })
```

Comment:
❓ How would you feel about moving this to a `context` method within the `SignIn::SessionContainer` model? It seems like everything we need to build the context can be found within a session container through attributes (`access_token`) or relationships (`credential_type` and `icn`).

Implementation might look something like this:

```ruby
module SignIn
  class SessionContainer
    def context
      access_token.to_s.merge(context_details_hash)
    end

    private

    def context_details_hash
      {
        type: session.user_verification.credential_type,
        icn: session.user_account.icn
      }
    end
  end
end
```

...and then you'd just adjust the sign in controller like so:

```ruby
session_container = SignIn::SessionRefresher.new(refresh_token: decrypted_refresh_token, anti_csrf_token:).perform
serializer_response = SignIn::TokenSerializer.new(session_container:, cookies: token_cookies).perform
sign_in_logger.info('refresh', session_container.context)
```

Reply:
Agreed, this is a lot cleaner!